src/context/AppProvider/AppContext.tsx

```diff
@@ -192,64 +190,121 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
           Object.assign(accountMetadataByAccountId, accountIdsAndMetadata)
 
           const { getAccount } = portfolioApi.endpoints
-          const accountPromises = accountIds.map(accountId =>
-            dispatch(getAccount.initiate({ accountId }, { forceRefetch: true })),
-          )
 
-          const accountResults = await Promise.allSettled(accountPromises)
+          // Chunks of AccountIds
+          // For UTXOs, each chain gets its own chunk to detect chain activity over other scriptTypes
+          // For others, no need to use chunks, but for the sake of consistency, we keep the same structure
+          const accountNumberAccountIdChunks = (
+            _accountIds: AccountId[],
+          ): Record<ChainId, AccountId[]> => {
+            return _accountIds.reduce(
+              (acc, _accountId) => {
+                const { chainId } = fromAccountId(_accountId)
+
+                if (isUtxoChainId(chainId)) {
+                  acc[chainId] = [_accountId]
+                  return acc
+                }
```

Comment:
[suggestion]: This path shouldnt be required if there arent multiple utxo account ids for a given chainId. I.e if there is no collision on chainId we can rely on the flow below this statement to correctly group accountIds by chainId - even if there is only 1 per chainId.

If the intent is to "overwrite" i.e "last write wins" then leave as-is

Reply:
You are absolutely right! Reading this again with fresh eyes, this looks blatantly wrong. A UTXO ChainId may and will most likely contain multiple AccountIds (one per scriptType), and we need them all in one "chunk" (terminology removed btw!) to detect activity across scriptTypes, that effectively voided it.
5c094cc272